src/clusterfuzz/_internal/tests/core/bot/tasks/trials_test.py

```diff
@@ -93,3 +111,87 @@ def test_selection_for_android_apk(self):
     trial_selector.setup_additional_args_for_app()
     self.assertEqual(environment.get_value('APP_ARGS'), '-x --a1')
     self.assertEqual(environment.get_value('TRIAL_APP_ARGS'), '--a1')
+
+  def test_no_effect_on_no_match_source_side(self):
+    """Ensure that no additional flags are added if a binary has no trials on source side."""
+    self.source_side_test(
+        """
+      [
+        {"app_args": "--c4", "app_name": "app_4", "probability": 0.2}
+      ]
+      """, 0.0, 'app_0', '-x', None)
```

Comment:
nit: Maybe just use Python list/dict here and turn into json inside the helper function? Might simplify readability.

Only on the last test it might not work to use invalid json. But maybe that test is not so useful after all. Or it could just have correct json with some missing keys?

Reply:
done